### PR TITLE
Add Support for multiple generators

### PR DESCRIPTION
**Pull request by PMunkes**
This pull request adds support for the generators GENIE, Tauola++ and PHOTOS++.
It also adds support for log4cpp, a dependency of GENIE.
It adds some PDF files to lhapdf5, that are needed by FairShip.
Finally it adds Support for MillepedeII.

### CODE REVIEW

**Comment by ktf** [comment]
Sorry for the late reply, looks reasonable to me. Can you please squash commits a bit? E.g. all the PHOTOS ones into one?

File: lhapdf5.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ rsync -a --exclude '**/.git' $SOURCEDIR/ ./
 make ${JOBS+-j $JOBS} all
 make install
 
-PDFSETS="cteq6l cteq6ll CT10 CT10nlo MSTW2008nnlo EPS09LOR_208 EPS09NLOR_208"
+PDFSETS="cteq6l cteq6ll CT10 CT10nlo MSTW2008nnlo EPS09LOR_208 EPS09NLOR_208 cteq66a cteq66a0 cteq4m"
```
Comment (ktf) [inline lhapdf5.sh]: How big are these? @qgp, any objection to include them?

File: pythia6.sh [hunk 2]
```diff
@@ -16,6 +16,11 @@ make ${JOBS+-j$JOBS}
 make install
 tar -c lib include | tar -x -C $INSTALLROOT 
 
+cd $INSTALLROOT/lib/
```
Comment (ktf) [inline pythia6.sh]: Why not simply:

```
ln -s libpythia6.so $INSTALLROOT/lib/libPythia6.so
```

File: millepedeii.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+version: V04-03-01
+source: https://github.com/PMunkes/MillepedeII
+tag: V04-03-01
+---
+#!/bin/bash
+rsync -a $SOURCEDIR/* .
+
+make  pede 
+./pede -t
+rsync pede $INSTALLROOT/bin/
```
Comment (ktf) [inline millepedeii.sh]: Why not `cp pede $INSTALLROOT/bin/`?

File: log4cpp.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+version: "%(tag_basename)s%(defaults_upper)s"
+tag: REL_1_1_1_Nov_26_2013
+source: https://github.com/PMunkes/log4cpp
+build_requires:
+  - autotools
+---
+#!/bin/bash -ex
+rsync -a $SOURCEDIR/* .
+./autogen.sh
+./configure          --prefix=$INSTALLROOT  \
```
Comment (ktf) [inline log4cpp.sh]: You should align `\`. If you use `vim` I suggest you install the [tabular](https://github.com/godlygeek/tabular) plugin or something similar.

File: photospp.sh [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  - pythia
+  - Tauolapp
+---
+#!/bin/sh
+
+export  HEPMCLOCATION="$HEPMC_ROOT"
+
+rsync -a $SOURCEDIR/* .
+
+./configure --enable-debug --with-hepmc=$HEPMC_ROOT --with-pythia8=$PYTHIA_ROOT --with-tauola=$TAUOLAPP_ROOT --prefix=$INSTALLROOT CFLAGS="$CFLAGS" CXXFLAGS="$CFLAGS"
```
Comment (ktf) [inline photospp.sh]: I suppose you did try running `$SOURCEDIR/configure`, right? Also, please use `rsync -a $SOURCEDIR/  .` (no `*`).

File: lhapdf5.sh (hunk 1 above)
Comment (PMunkes) [inline lhapdf5.sh]: These PDF-files are approximately 500 kB in size.

**Comment by PMunkes** [comment]
I have rebased all of the commits and adressed most of the comments. At the moment there is only one blocker from my side, most of the repositories for these recipes are still on my Github account. Once they have been transferred, I would need to change all the URLs from ../PMunkes/.. to ../ShipSoft/... . Until then I am submitting these changes for review.

**Comment by ktf** [comment]
I think these have been moved to `shipdist`.

**Pull request closed without merging**